runtime/src/main/java/org/corfudb/runtime/collections/StreamManager.java

```diff
@@ -123,6 +123,33 @@ void subscribe(@Nonnull StreamListener streamListener, @Nonnull String namespace
         subscriptions.put(streamListener, new StreamSubscriber(task, scheduledFuture));
     }
 
+    /**
+     * Subscribe to updates.
+     *
+     * @param streamListener   client listener
+     * @param namespace        namespace of interest
+     * @param streamTag        only updates of tables with the stream tag will be polled
+     * @param tablesOfInterest only updates from these tables will be returned
+     * @param startAddress     address to start the notifications from
+     */
+    public synchronized <K extends Message, V extends Message, M extends Message>
+    void subscribe(@Nonnull StreamListener streamListener, @Nonnull String namespace,
+                   @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest,
+                   long startAddress) {
+        // The table should be opened full schema before subscription.
+        TableRegistry registry = runtime.getTableRegistry();
+        List<TableSchema<K, V, M>> tableSchemas = tablesOfInterest
+                .stream()
+                .map(tName -> {
+                    Table<K, V, M> table = registry.getTable(namespace, tName);
```

Comment:
If a table has not been opened, this would be null and line 145 would hit a null pointer exception right?
Do you want to throw an exception instead?

Reply:
@hisundar If table is not opened, getTable() will throw NoSuchElementException or IllegalArgumentException

Reply:
Right, so can we just mention that in the java doc of this method?